controllers/services/auth/auth_controller_actions.go

```diff
@@ -0,0 +1,233 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package auth
+
+import (
+	"context"
+	"errors"
+	"slices"
+	"strings"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	k8serr "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/apis/services/v1alpha1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Templates = []odhtypes.TemplateInfo{
+		{
+			FS:   resourcesFS,
+			Path: AdminGroupRoleTemplate,
+		},
+		{
+			FS:   resourcesFS,
+			Path: AllowedGroupRoleTemplate,
+		},
+	}
+
+	return nil
+}
+
+func copyGroups(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	ai, ok := rr.Instance.(*serviceApi.Auth)
+	if !ok {
+		return errors.New("instance is not of type *services.Auth")
+	}
+
+	newGroups := false
+
+	// Get groups from the dashboardConfig
+	odhObject := &unstructured.Unstructured{}
+	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
+
+	err := rr.Client.Get(ctx, client.ObjectKey{
+		Name:      "odh-dashboard-config",
+		Namespace: rr.DSCI.Spec.ApplicationsNamespace,
+	}, odhObject)
+	if err == nil {
+		// We found the dashboardConfig
+		// If there is an error here we are ignoring it because we don't need the
+		// dashboardConfig to exist
+		foundGroups, found, _ := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
+		if !found {
+			return errors.New("no groupsConfig found in dashboardConfig")
+		}
+
+		adming := foundGroups["adminGroups"]
+		if len(adming) != 0 {
+			for _, e := range strings.Split(adming, ",") {
+				e = strings.TrimSpace(e)
+				// If the group is already in the list then ignore it
+				if !slices.Contains(ai.Spec.AdminGroups, e) {
+					ai.Spec.AdminGroups = append(ai.Spec.AdminGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+
+		allowedg := foundGroups["allowedGroups"]
+		if len(allowedg) != 0 {
+			for _, e := range strings.Split(allowedg, ",") {
+				// If the group is already in the list then ignore it
+				if !slices.Contains(ai.Spec.AllowedGroups, e) {
+					ai.Spec.AllowedGroups = append(ai.Spec.AllowedGroups, strings.TrimSpace(e))
+					newGroups = true
+				}
+			}
+		}
+	}
+	if err != nil && !k8serr.IsNotFound(err) {
```

Comment:
it does not change the logic so you can leave it as it is, but maybe this can be done just after the `Get`, so you can avoid the `if err == nil`

Reply:
:+1: